src/conn/conn_sweep.c

```diff
@@ -45,6 +47,30 @@ __sweep_mark(WT_SESSION_IMPL *session, uint64_t now)
           __wt_atomic_loadi32(&dhandle->session_inuse) > 0 || dhandle->timeofdeath != 0)
             continue;
 
+        /*
+         * The sweep server's algorithm is altered to prevent unnecessary table dhandle closures.
+         * This is done by checking for associated file dhandles before marking table dhandles for
+         * sweeping. It resolves schema lock contention caused by repetitive table dhandle
+         * operations during MongoDB cursor activity on simple tables, and ensures table dhandles
+         * are retained for active file dhandles, which is required for file dhandle access.
+         */
+        if (dhandle->type == WT_DHANDLE_TYPE_TABLE) {
+            table = (WT_TABLE *)dhandle;
+            if (table->is_simple && table->cgroups != NULL) {
+                const char *uri = table->cgroups[0]->source;
```

Comment:
Do we really need this local variable? we are not using it anywhere else.

Reply:
@kommiharibabu Fixed in d03558040e5382971e61a3cff5c5ad33c53db828